otelconf/config.go

```diff
@@ -0,0 +1,171 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otelconf provides an OpenTelemetry declarative configuration SDK.
+package otelconf // import "go.opentelemetry.io/contrib/otelconf"
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/otel/log"
+	nooplog "go.opentelemetry.io/otel/log/noop"
+	"go.opentelemetry.io/otel/metric"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+	yaml "go.yaml.in/yaml/v3"
+
+	"go.opentelemetry.io/contrib/otelconf/internal/provider"
+)
+
+// SDK is a struct that contains all the providers
+// configured via the configuration model.
+type SDK struct {
+	meterProvider  metric.MeterProvider
+	tracerProvider trace.TracerProvider
+	loggerProvider log.LoggerProvider
+	shutdown       shutdownFunc
+}
+
+// TracerProvider returns a configured trace.TracerProvider.
+func (s *SDK) TracerProvider() trace.TracerProvider {
+	return s.tracerProvider
+}
+
+// MeterProvider returns a configured metric.MeterProvider.
+func (s *SDK) MeterProvider() metric.MeterProvider {
+	return s.meterProvider
+}
+
+// LoggerProvider returns a configured log.LoggerProvider.
+func (s *SDK) LoggerProvider() log.LoggerProvider {
+	return s.loggerProvider
+}
+
+// Shutdown calls shutdown on all configured providers.
+func (s *SDK) Shutdown(ctx context.Context) error {
+	return s.shutdown(ctx)
+}
+
+var noopSDK = SDK{
+	loggerProvider: nooplog.LoggerProvider{},
+	meterProvider:  noopmetric.MeterProvider{},
+	tracerProvider: nooptrace.TracerProvider{},
+	shutdown:       func(context.Context) error { return nil },
+}
+
+// NewSDK creates SDK providers based on the configuration model.
+func NewSDK(opts ...ConfigurationOption) (SDK, error) {
+	o := configOptions{
+		ctx: context.Background(),
+	}
+	for _, opt := range opts {
+		o = opt.apply(o)
+	}
+	if o.opentelemetryConfig.Disabled != nil && *o.opentelemetryConfig.Disabled {
+		return noopSDK, nil
+	}
+
```

Comment:
Should we setup the SDK internal logger by the value of `log_level`?

Reply:
Happy to follow up this PR with another one to enable the internal logger